sources/api/netdog/src/cli/install.rs

```diff
@@ -54,14 +51,57 @@ pub(crate) fn run(args: InstallArgs) -> Result<()> {
             interface_type @ (InterfaceType::Dhcp | InterfaceType::Static),
             InterfaceFamily::Ipv4 | InterfaceFamily::Ipv6,
         ) => {
-            let lease = fetch_lease(primary_interface, interface_type, args.data_file)?;
+            let lease = fetch_lease(&primary_interface, interface_type, args.data_file)?;
             write_resolv_conf(&lease)?;
             write_current_ip(&lease.ip_address.addr())?;
+
+            // If we haven't already, set and apply default sysctls for the primary network
+            // interface
+            if !Path::exists(Path::new(PRIMARY_SYSCTL_CONF)) {
+                write_interface_sysctl(primary_interface, PRIMARY_SYSCTL_CONF)?;
+
+                // Execute `systemd-sysctl` with our configuration file to set the sysctls
+                let systemd_sysctl_result = Command::new(SYSTEMD_SYSCTL)
+                    .arg(PRIMARY_SYSCTL_CONF)
+                    .output()
+                    .context(error::SystemdSysctlExecutionSnafu)?;
+                ensure!(
+                    systemd_sysctl_result.status.success(),
+                    error::FailedSystemdSysctlSnafu {
+                        stderr: String::from_utf8_lossy(&systemd_sysctl_result.stderr)
+                    }
+                );
```

Comment:
If for whatever reason we are able to write the sysctl file, but the `systemd-sysctl` command fails, we'll never retry applying the settings on a second DHCP lease event.

It might be better to separate these steps:
* write the sysctl.conf if it doesn't exist
* run the systemd-sysctl command if a specific marker file doesn't exist, then write the marker file